src/python/foglamp/device/ingest.py

```diff
@@ -53,62 +55,185 @@ class Ingest(object):
     _discarded_readings = 0  # type: int
     """Number of readings rejected before statistics were flushed to storage"""
 
-    _write_statistics_loop_task = None  # type: asyncio.Task
-    """asyncio task for :meth:`_write_statistics_loop`"""
+    _write_statistics_task = None  # type: asyncio.Task
+    """asyncio task for :meth:`_write_statistics`"""
 
-    _sleep_task = None  # type: asyncio.Task
+    _write_statistics_sleep_task = None  # type: asyncio.Task
     """asyncio task for asyncio.sleep"""
 
-    _stop = False  # type: bool
+    _stop = False
     """Set to true when the server needs to stop"""
 
+    _started = False
+    """True when the server has been started"""
+
+    _engine = None  # type: aiopg.sa.Engine
+    """Database connection pool"""
+
+    _insert_queue = None  # type: asyncio.Queue
+    """insert objects are added to this queue"""
+
+    _insert_tasks = None  # type: List[asyncio.Task]
+    """asyncio tasks for :meth:`_insert_readings`"""
+
+    _get_next_insert_tasks = None  # type: List[asyncio.Task]
+    """asyncio tasks for asyncio.Queue.get called by :meth:`_insert_readings`"""
+
+    # Configuration
+    _max_db_connections = 4
```

Comment:
Ideally we should make this configurable so that we can easily tune this when doing performance tests